lib/clean.py

```diff
@@ -691,3 +691,33 @@ def strip_birth_date(df: pd.DataFrame, cols: list[str]) -> pd.DataFrame:
     for col in cols:
         df.loc[:, col] = strip_birth_day_and_month(df[col])
     return df
+
+
+def canonicalize_names(
+    df: pd.DataFrame,
+    clusters: list[tuple],
+    uid_column: str = "uid",
+    first_name_column: str = "first_name",
+    last_name_column: str = "last_name",
+) -> pd.DataFrame:
+    for cluster in clusters:
+        uid, first_name, last_name = None, "", ""
+        for idx in cluster:
+            row = df.loc[df[uid_column] == idx]
+            if (
+                uid is None
+                or len(row[first_name_column]) > len(first_name)
+                or (
+                    len(row[first_name_column]) == len(first_name)
+                    and len(row[last_name_column]) > len(last_name)
+                )
+            ):
+                uid, first_name, last_name = (
+                    idx,
+                    row[first_name_column],
+                    row[last_name_column],
+                )
+        df.loc[df[uid_column].isin(cluster), uid_column] = uid
+        df.loc[df[uid_column] == uid, first_name_column] = first_name
+        df.loc[df[uid_column] == uid, last_name_column] = last_name
```

Comment:
I think the last 2 rows should be:
```python
        df.loc[df[uid_column].isin(cluster), first_name_column] = first_name
        df.loc[df[uid_column].isin(cluster), last_name_column] = last_name
```
The original seems to be no op.

Reply:
Nice. I made that change when I was experimenting, but reverted back to what you originally produced.